paywall/src/data-iframe/Mailbox.ts

```diff
@@ -316,6 +320,33 @@ export default class Mailbox {
     }
   }
 
+  /**
+   * When we receive PostMessages.SET_USER_METADATA, simply forward it
+   * to the handler
+   */
+  setUserMetadata = async ({
+    lockAddress,
+    metadata,
+  }: {
+    lockAddress: string
+    metadata: UserMetadata
+  }) => {
+    if (!this.handler) {
+      this.emitError(
+        new Error('blockchain handler not instantiated, cannot set metadata')
+      )
+      return
+    }
+
+    try {
+      await this.handler.setUserMetadata(lockAddress, metadata)
+      // success, no error returned
+      this.postMessage(PostMessages.SET_USER_METADATA_SUCCESS, undefined)
+    } catch (error) {
+      this.emitError(error)
```

Comment:
There is already a generic `PostMessages.ERROR` that is used for failures, and I opted to continue using it here for consistency, so there is no specific metadata error postmessage.